File: java/src/main/scala/io/taskmonk/entities/Task.scala
package io.taskmonk.entities

import java.util
import java.util.Date

import play.api.libs.json.Json

import scala.collection.JavaConverters._


object Task {

  implicit val reads = Json.reads[Task]
  implicit val writes = Json.writes[Task]
}
case class Task(externalId: String,
                project_id: String,
                batch_id: String,
                input: Map[String, String],
                output: Map[String, String]) {
  def this(externalId: String,
            project_id: String,
            batch_id: String,
            input: java.util.Map[String, String]) = {
    this(externalId, project_id, batch_id, input.asScala.toMap, Map.empty[String, String])
  }
  def getInput(): java.util.Map[String, String] = {
    input.asJava
  }
  def getOutput(): java.util.Map[String, String] = {
    output.asJava
  }
  def getExternalId(): String = {
    externalId
  }
}

File: java/src/main/scala/io/taskmonk/entities/Task.java
package io.taskmonk.entities;

import java.util.HashMap;
import java.util.Map;

public class Task {
    public String externalId;
    public String projectId;
    public String batchId;
    public Map<String, String> input;
    public Map<String, String> output = new HashMap<String, String>();

    public Task(String externalId, String projectId, String batchId,
                Map<String, String> input) {
        this.externalId = externalId;
        this.projectId = projectId;
        this.batchId = batchId;
        this.input = input;
    }
}
